torch/_dynamo/variables/nn_module.py

```diff
@@ -283,37 +282,53 @@ def get_kwargs(*names):
             bound_args = bound_args.arguments
             return {k: bound_args[k] for k in names}
 
-        def wrap_values(items, getsource=AttrSource):
+        def wrap_values(items, is_parameter=False):
             result = []
             for name, submod in items:
-                # layer.0.foo => layer[0].foo
-                name = re.sub(r"[.]([0-9]+)([.]|$)", r"[\1]\2", name)
-                src = NNModuleSource(getsource(self.source, name))
                 result.append(
                     tx.output.register_attr_or_module(
                         submod,
                         key,
                         name,
-                        source=src,
+                        source=NNModuleSource(
+                            gen_source(self.source, name, is_parameter)
+                        ),
                         **options,
                     )
                 )
             return ListIteratorVariable(result, mutable_local=MutableLocal(), **options)
 
-        def named_embed(name, obj):
+        def named_embed(name, obj, is_parameter=False):
             return TupleVariable(
                 [
                     ConstantVariable(name, **options),
                     tx.output.register_attr_or_module(
                         obj,
                         key,
                         name,
-                        source=NNModuleSource(GetItemSource(self.source, name)),
+                        source=NNModuleSource(
+                            gen_source(self.source, name, is_parameter)
+                        ),
                         **options,
                     ),
                 ]
             )
 
+        def gen_source(source, name, is_parameter=False):
+            name_split = name.split(".")
+            if name_split[0] == "":
+                return source
+            while len(name_split) > 1:
+                x = name_split.pop(0)
+                source = AttrSource(source, x)
+
+            if is_parameter:
+                return GetItemSource(
+                    AttrSource(source, "_parameters"), name_split.pop(0)
```

Comment:
Direct access to `_parameters` doesn't smell right

Reply:
Thanks for catching this! Yes, I found we can directly use ```.weights``` to access, so the code is more concise now. In this PR, we only need to fix the bug of the nested module architecture reconstruction.